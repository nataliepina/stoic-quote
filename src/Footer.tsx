import React from "react";

const Footer = () => {
  const copyright: string = "Natalie Pina &copy;2021";

  return (
    <footer className="footer">
      <p>
        <a
          data-testid="footer"
          href="https://github.com/nataliepina/stoic-quote"
          target="_blank"
          rel="noopener noreferrer"
        >
          {copyright}
        </a>
      </p>
    </footer>
  );
};

export default Footer;
