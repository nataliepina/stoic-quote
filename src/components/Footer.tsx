import React from 'react';

const Footer: React.FC = (): JSX.Element => {
  const copyright = 'Natalie Pina \u00A9 2021';

  return (
    <footer className="footer">
      <p>
        <a
          data-testid="footer"
          href="https://github.com/nataliepina/stoic-quote"
          target="_blank"
          rel="noopener noreferrer"
        >
          {copyright}
        </a>
      </p>
    </footer>
  );
};

export default Footer;
