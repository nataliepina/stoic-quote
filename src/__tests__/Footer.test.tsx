import { render } from '@testing-library/react';
import React from 'react';
import Footer from '../components/Footer';

describe('Footer tests', () => {
  it('should render without breaking', () => {
    render(<Footer />);
  });

  it('should match snapshot', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render footer text', () => {
    const { getByTestId } = render(<Footer />);
    expect(getByTestId('footer')).toHaveTextContent('Natalie Pina');
    expect(getByTestId('footer')).toHaveAttribute(
      'href',
      'https://github.com/nataliepina/stoic-quote'
    );
  });
});
