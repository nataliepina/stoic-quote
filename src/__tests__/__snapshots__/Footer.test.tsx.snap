// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer tests should match snapshot 1`] = `
<DocumentFragment>
  <footer
    class="Footer"
  >
    <p>
      <a
        data-testid="footer"
        href="https://github.com/nataliepina"
        rel="noopener noreferrer"
        target="_blank"
      >
        Natalie Pina Â©2021
      </a>
    </p>
  </footer>
</DocumentFragment>
`;
